parodos-dev/parodos
# FLPATH-481: Fix maven build order

**What this PR does / why we need it**:
The current Maven build order for our project is causing an issue where the `openapi.json` file, generated by the `notification-service` module, is not generated in time. The issue arises because the `notification-service-sdk` module is built before the notification-service module, which is required for generating the `openapi.json` file.

This PR addresses the problem by removing the dependency of `parodos-model-api` on `notification-service-sdk`.
After the changes, the build order is adjusted to ensure that the `notification-service` module is built before the `notification-service-sdk` module, allowing the `openapi.json` file to be generated correctly.

The current Maven build order is:
```
[INFO] Reactor Build Order:
[INFO] 
[INFO] parodos                                                            [pom]
[INFO] workflow-engine                                                    [jar]
[INFO] notification-service-sdk                                           [jar]
[INFO] parodos-model-api                                                  [jar]
[INFO] notification-service                                               [jar]
[INFO] prebuilt-tasks                                                     [jar]
[INFO] workflow-service                                                   [jar]
[INFO] workflow-service-sdk                                               [jar]
[INFO] sdk-utils                                                          [jar]
[INFO] workflow-examples                                                  [jar]
[INFO] pattern-detection-library                                          [jar]
[INFO] coverage                                                           [jar]
[INFO] integration-tests                                                  [jar]
``` 

The updated Maven build order is:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] parodos                                                            [pom]
[INFO] workflow-engine                                                    [jar]
[INFO] parodos-model-api                                                  [jar]
[INFO] notification-service                                               [jar]
[INFO] notification-service-sdk                                           [jar]
[INFO] prebuilt-tasks                                                     [jar]
[INFO] workflow-service                                                   [jar]
[INFO] workflow-service-sdk                                               [jar]
[INFO] sdk-utils                                                          [jar]
[INFO] workflow-examples                                                  [jar]
[INFO] pattern-detection-library                                          [jar]
[INFO] coverage                                                           [jar]
[INFO] integration-tests                                                  [jar]
```

**How to test**:
1 - Delete `notification-service/generated/openapi.json`
2 - Run `mvn clean install -DskipTest `
3 - Verify the Maven build order

**Which issue(s) this PR fixes**:
Fixes # [FLPATH-481](https://issues.redhat.com/browse/FLPATH-481)

**Change type**
- [ ] Auto-generated SDK code

**Impacted services**
- [ ] Workflow Service

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
